### ISSUE TITLE
`Bug` changine wifi details OTA

### ISSUE THREAD

**Issue by geekyclarkey** [Blackymas/NSPanel_HA_Blueprint#2302]
### Panel Model

EU

### What is the bug?

Hi, you have set the ota password the same as the wifi passsword. So now i have changed my wifi, i cant input the new details and do an update OTA. As the new wifi password is different. it says mentions authentication issue when trying to upload it.
can you either not use an ota password or add it into the substitutions so we can decide what its going to be?

The only way around this at the minute is to tft flash all my panels again. but OTA would be far easier.

### Your Panel's YAML

```yaml
substitutions:
  ###### CHANGE ME START ######
  device_name: "theospanel" 
  wifi_ssid: "hallwifi"
  wifi_password: "1234554321"
  ota_password: "whjbw78238edsjnx98whdaa"

  nextion_update_url: "http://192.168.100.229:8123/local/nspanel_eu.tft"

  ##### addon-configuration #####
  ## addon_climate ##
  # addon_climate_heater_relay: "1" # possible values: 1/2

  ##### CHANGE ME END #####

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml # Core package
      # - nspanel_esphome_advanced.yaml # activate advanced (legacy) elements - can be useful for troubleshooting
      # - nspanel_esphome_addon_climate_cool.yaml # activate for local climate (cooling) control
      # - nspanel_esphome_addon_climate_heat.yaml # activate for local climate (heater) control
    refresh: 300s

##### My customization - Start #####
##### My customization - End #####
```

**Comment by edwardtfn** [comment]
Substitutions just like in your example should work. Have you tried that?

**Comment by geekyclarkey** [comment]
Hi, to be honest no I haven’t tried. Do the substitutions I put in my local esp home file override the esphome core file it uses when compiling? Sent from my iPhoneOn 30 Sep 2024, at 07:34, Edward Firmo ***@***.***> wrote:﻿
Substitutions just like in your example should work. Have you tried that?

**Comment by edwardtfn** [comment]
In fact this use of Wi-Fi password as OTA password was in this project even before I got involved on it. I never liked that much, but they probably had a reason for setting up like that. And it's too painful to change that as it will break to all users.

Anyhow, some versions ago I've moved some of those things to their own substitutions, so OTA password uses a substitution named `ota_password` as defined here:

https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/e146b1f35dfe38cd8d2da90828695f8a37865b0c/esphome/nspanel_esphome_core.yaml#L136-L140

And then, to avoid breaking all the existing installations, there is a substitution defining this as the Wi-Fi password:

https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/e146b1f35dfe38cd8d2da90828695f8a37865b0c/esphome/nspanel_esphome_core.yaml#L8-L19

As ESPHome compiler will use the remote substitution only when a local equivalent substitution is not available, if you define a substitution locally as you have done, it will use that, otherwise, it will use the `wifi_password` for OTA password.

**Comment by geekyclarkey** [comment]
Hi,Yea that worked perfectly. I just added ota_password to the substitutions and set it to my old Wi-Fi password. Thanks for your help. I love this project and have a lot of respect and gratitude to you and all of the contributors for all of your hard work. Many thanks Sean.Sent from my iPhoneOn 30 Sep 2024, at 13:17, Edward Firmo ***@***.***> wrote:﻿
In fact this use of Wi-Fi password as OTA password was in this project even before I got involved on it. I never liked that much, but they probably had a reason for setting up like that. And it's too painful to change that as it will break to all users.
Anyhow, some versions ago I've moved some of those things to their own substitutions, so OTA password uses a substitution named ota_password as defined here:
https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/e146b1f35dfe38cd8d2da90828695f8a37865b0c/esphome/nspanel_esphome_core.yaml#L136-L140
And then, to avoid breaking all the existing installations, there is a substitution defining this as the Wi-Fi password:
https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/e146b1f35dfe38cd8d2da90828695f8a37865b0c/esphome/nspanel_esphome_core.yaml#L8-L19
As ESPHome compiler will use the remote substitution only when a local equivalent substitution is not available, if you define a substitution locally as you have done, it will use that, otherwise, it will use the wifi_password for OTA password.
